src/docfx/SubCommands/BuildCommand.cs

```diff
@@ -307,45 +301,72 @@ private static void MergeGitContributeToConfig(BuildJsonConfig config)
             }
         }
 
-        internal static Dictionary<string, FileMetadataPairs> GetFileMetadataFromOption(ListWithStringFallback fileMetadataFilePaths, Dictionary<string, FileMetadataPairs> fileMetadataFromConfig)
+        internal static Dictionary<string, FileMetadataPairs> GetFileMetadataFromOption(ListWithStringFallback fileMetadataFilePaths, string fileMetadataFilePath, Dictionary<string, FileMetadataPairs> fileMetadataFromConfig)
         {
             var fileMetadataFromFile = new Dictionary<string, FileMetadataPairs>();
             if (fileMetadataFilePaths != null)
             {
-                foreach (var fileMetadataFilePath in fileMetadataFilePaths)
+                foreach (var filePath in fileMetadataFilePaths)
                 {
                     Dictionary<string, FileMetadataPairs> metadata = null;
                     try
                     {
-                        metadata = JsonUtility.Deserialize<BuildJsonConfig>(fileMetadataFilePath).FileMetadata;
+                        metadata = JsonUtility.Deserialize<Dictionary<string, FileMetadataPairs>>(filePath);
                     }
                     catch (FileNotFoundException)
                     {
-                        Logger.LogWarning($"Invalid option \"{fileMetadataFilePath}\": file does not exist, ignored.");
+                        Logger.LogWarning($"Invalid option \"{filePath}\": file does not exist, ignored.");
                     }
                     catch (JsonException e)
                     {
-                        Logger.LogWarning($"File from \"{fileMetadataFilePath}\" is not a valid JSON format file metadata, ignored: {e.Message}");
+                        Logger.LogWarning($"File from \"{filePath}\" is not a valid JSON format file metadata, ignored: {e.Message}");
                     }
                     if (metadata == null)
                     {
-                        Logger.LogWarning($"File from \"{fileMetadataFilePath}\" does not contain \"fileMetadata\" definition, ignored.");
+                        Logger.LogWarning($"File from \"{filePath}\" does not contain \"fileMetadataPairs\" definition, ignored.");
                     }
                     else
                     {
                         fileMetadataFromFile = MergeDictionary(
```

Comment:
>MergeDictionary [](start = 47, length = 15)

what about following case:
FileA: apply metadata X = 1 to a/**
FileB: apply metadata X = 2 to b/**